ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fullsync/FullImportBlockStep.java

```diff
@@ -35,18 +35,25 @@
   private final EthContext ethContext;
   private long gasAccumulator = 0;
   private long lastReportMillis = 0;
+  private final FullSyncTerminationCondition fullSyncTerminationCondition;
 
   public FullImportBlockStep(
       final ProtocolSchedule protocolSchedule,
       final ProtocolContext protocolContext,
-      final EthContext ethContext) {
+      final EthContext ethContext,
+      final FullSyncTerminationCondition fullSyncTerminationCondition) {
     this.protocolSchedule = protocolSchedule;
     this.protocolContext = protocolContext;
     this.ethContext = ethContext;
+    this.fullSyncTerminationCondition = fullSyncTerminationCondition;
   }
 
   @Override
   public void accept(final Block block) {
+    if (fullSyncTerminationCondition.shouldStopDownload()) {
+      LOG.info("Not importing another block, because terminal condition was reached.");
```

Comment:
This guy needs to be DEBUG.  Running on devnet4 the logs are getting spammed for every block post TTD until the pipeline finishes (at the current peers' head).

maybe this could be emitted at info level if `block.number() mod 100 == 0 ` ?

Reply:
Ok I'll make it debug.